sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_internal/__init__.py

```diff
@@ -60,12 +56,22 @@ def parse_blob_storage_url(blob_storage_url):
     """
 
     try:
-        folder_name = blob_storage_url.rstrip("/").split("/")[-1]
-        container_url = blob_storage_url[: blob_storage_url.rindex(folder_name) - 1]
+        parsed = urlparse(folder_url)
+
+        # the first segment of the path is the container name
+        container = parsed.path.strip("/").split("/")[0]
+
+        # N.B. this discards any SAS token in the URL.
+        # This is intentional--client methods require the SAS token as a separate parameter.
+        container_url = "{}://{}/{}".format(parsed.scheme, parsed.netloc, container)
+
+        # the folder name is the rest of the path
+        folder_name = parsed.path[len(container) + 1 :].strip("/")
```

Comment:
Nit: final strip seems unnecessary since you did it above and started beyond the container's slash.

Reply:
`parsed.path` has a leading slash. I refactored this to be clearer.